### PR TITLE
Pass -Wunguarded-availability-new to quiet a new clang warning

### PR DESCRIPTION
**Pull request by chamons**
- https://bugzilla.xamarin.com/show_bug.cgi?id=58574

### CODE REVIEW

**Comment by chamons** [comment]
I added it to xcode9 instead of master since it only pops up for now w\ the new clang.

I can commit in master then cherry pick commit if needed, but I was concerned about merge issues.

**Comment by rolfbjarne** [comment]
We should probably do this for mtouch as well.

**Comment by chamons** [comment]
@rolfbjarne Let me add that in this PR shortly.

File: tools/mmp/driver.cs [hunk 1]
```diff
@@ -1251,6 +1251,7 @@ static int Compile ()
 					args.Append ("-fobjc-runtime=macosx ");
 				if (!embed_mono)
 					args.Append ("-DDYNAMIC_MONO_RUNTIME ");
+				args.Append ("-Wno-unguarded-availability-new ");
```
Comment (spouliot) [inline tools/mmp/driver.cs]: does older `clang` versions support it ?
or are we trading this warning for another ?

Comment (chamons) [inline tools/mmp/driver.cs]: Good catch.

with 8.3
``` xcrun clang -Wno-unguarded-availability-new for.c
warning: unknown warning option '-Wno-unguarded-availability-new'; did you mean '-Wno-unguarded-availability'? [-Wunknown-warning-option]
1 warning generated.
```

with 9b4
```$ xcrun clang -Wno-unguarded-availability-new for.c```

Comment (chamons) [inline tools/mmp/driver.cs]: Added a version check.

**Comment by chamons** [comment]
@spouliot 👍 ?

**Pull request merged**